sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -434,6 +434,248 @@ static int hda_ipc4_post_trigger(struct snd_sof_dev *sdev, struct snd_soc_dai *c
 	return ret;
 }
 
+static int hda_dai_set_up_widgets_in_pipeline(struct snd_soc_dapm_widget *w,
+					      struct snd_sof_pipeline *spipe, int dir)
+{
+	struct snd_sof_widget *swidget = w->dobj.private;
+	struct snd_sof_dev *sdev = widget_to_sdev(w);
+	struct snd_soc_dapm_path *p;
+	int ret;
+
+	/* set up the widget if it belongs to the same pipeline as the DAI widget */
+	if (swidget->spipe != spipe || !swidget->prepared)
```

Comment:
Do we need to go through the widgets in the path like hda_dai_unprepare_widgets_in_pipeline() does in the `!swidget->prepared` case? And same question for hda_dai_free_widgets_in_pipeline().